test/blackbox-tests/test-cases/enabled_if/eif-library-name-collision.t

```diff
@@ -0,0 +1,103 @@
+Showcase that using same library name in two contexts is possible for private libraries
+
+  $ mkdir -p a b
+  $ cat > dune-project << EOF
+  > (lang dune 3.13)
+  > (package (name bar) (allow_empty))
+  > (package (name baz) (allow_empty))
+  > EOF
+
+  $ cat > dune-workspace << EOF
+  > (lang dune 3.13)
+  > 
+  > (context default)
+  > 
+  > (context
+  >  (default
+  >   (name alt-context)))
+  > EOF
+  $ cat > a/dune << EOF
+  > (library
+  >  (name foo)
+  >  (enabled_if (= %{context_name} "default")))
+  > EOF
+  $ cat > a/foo.ml <<EOF
+  > let x = "foo"
+  > EOF
+
+  $ cat > b/dune << EOF
+  > (library
+  >  (name foo)
+  >  (enabled_if (= %{context_name} "alt-context")))
+  > EOF
+  $ cat > b/foo.ml <<EOF
+  > let x = "foo"
+  > EOF
+
+  $ dune build
+
+But not for public libraries
+
+  $ cat > a/dune << EOF
+  > (library
+  >  (name foo)
+  >  (public_name bar.foo)
+  >  (enabled_if (= %{context_name} "default")))
+  > EOF
+
+  $ cat > b/dune << EOF
+  > (library
+  >  (name foo)
+  >  (public_name baz.foo)
+  >  (enabled_if (= %{context_name} "alt-context")))
+  > EOF
+
+  $ dune build
+  Error: Library foo is defined twice:
```

Comment:
I have started an attempt in https://github.com/ocaml/dune/pull/9839/commits/619ea9dfccde0057dad3eaf0ee34346ec480e42d.

This doesn't work yet, as it seems the result from querying the DB to find a public library is ignored. There's a call to `find` here:

https://github.com/ocaml/dune/blob/98c576e643450659e2893ced9ae8e076b6129d95/src/dune_rules/scope.ml#L392

But this function already skips all invalid results to just return `None`:

https://github.com/ocaml/dune/blob/98c576e643450659e2893ced9ae8e076b6129d95/src/dune_rules/lib.ml#L1866

And then back in the call point, there's a comment that shows that somehow this should be dealt with:

https://github.com/ocaml/dune/blob/98c576e643450659e2893ced9ae8e076b6129d95/src/dune_rules/scope.ml#L395-L397

It seems if the result is `Invalid` (because there are 2 libraries defined) we should do something else?